File: CandyGame/CandyGame.py
import os, time, random, sys, select

width, height = 30, 15
box_x, box_w = width//2, 6
candy_x, candy_y = random.randint(0, width-1), 0
score, life = 0, 5

def clear():
    os.system("cls" if os.name == "nt" else "clear")

def key_pressed():
    return select.select([sys.stdin], [], [], 0)[0]

while life > 0:
    clear()
    for y in range(height):
        line = ""
        for x in range(width):
            if y == candy_y and x == candy_x:
                line += "O"
            elif y == height-2 and box_x <= x < box_x + box_w:
                line += "#"
            else:
                line += " "
        print(line)
    print(f"Score: {score}  Life: {life}  (a=left, d=right, q=quit)")

    start = time.time()
    while time.time() - start < 0.25:
        if key_pressed():
            ch = sys.stdin.read(1)
            if ch == "q":
                sys.exit()
            if ch == "a" and box_x > 0:
                box_x -= 1
            if ch == "d" and box_x + box_w < width:
                box_x += 1

    candy_y += 1
    if candy_y >= height-2:
        if box_x <= candy_x < box_x + box_w:
            score += 10
        else:
            life -= 1
        candy_y = 0
        candy_x = random.randint(0, width-1)

clear()
print("GAME OVER! Final Score:", score)

File: CandyGame/CandyGame.java
import java.util.*;

public class CandyGame {
    static int width = 30;
    static int height = 15;
    static int boxX = width / 2, boxW = 6;
    static int candyX = new Random().nextInt(width), candyY = 0;
    static int score = 0, life = 5;

    static void clear() { System.out.print("\033[H\033[2J"); System.out.flush(); }

    static void draw() {
        for (int y = 0; y < height; y++) {
            for (int x = 0; x < width; x++) {
                if (y == candyY && x == candyX) System.out.print("O");
                else if (y == height - 2 && x >= boxX && x < boxX + boxW) System.out.print("#");
                else System.out.print(" ");
            }
            System.out.println();
        }
        System.out.println("Score: " + score + "  Life: " + life + "  (a=left, d=right, q=quit)");
    }

    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        Random r = new Random();

        while (life > 0) {
            clear();
            draw();

            long start = System.currentTimeMillis();
            while (System.currentTimeMillis() - start < 250) {
                if (System.in.available() > 0) {
                    char ch = (char) System.in.read();
                    if (ch == 'q') return;
                    if (ch == 'a' && boxX > 0) boxX--;
                    if (ch == 'd' && boxX + boxW < width) boxX++;
                }
            }

            candyY++;
            if (candyY >= height - 2) {
                if (candyX >= boxX && candyX < boxX + boxW) score += 10;
                else life--;
                candyY = 0;
                candyX = r.nextInt(width);
            }
        }
        clear();
        System.out.println("GAME OVER! Final Score: " + score);
    }
}
